### PR TITLE
provide access to original model (ref #419)

### PR DESCRIPTION
**Pull request by davincho**

### CODE REVIEW

**Comment by kentcdodds** [comment]
Fantastic! Thanks for writing this up! I hope you don't mind iterating on this a little bit. I have a comment or two. Thanks!

File: src/directives/formly-field.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
     scope: {
       options: '=',
       model: '=',
+      originalModel: '=',
```
Comment (kentcdodds) [inline src/directives/formly-field.js]: Could you make this an optional attribute by changing `=` to `=?` please? And then in the controller (right in [here](https://github.com/davincho/angular-formly/blob/master/src/directives/formly-field.js#L87-L92)), could you add a default for that value? Something like:

```javascript
formlyUtil.reverseDeepMerge(options, {
  originalModel: options.model,
  extras: {},
  data: {},
  templateOptions: {},
  validation: {}
});
```

So if the `original-model` isn't specified (in the case of someone using the directive directly), it defaults to the given model.

And then we'd need a test to verify this defaulting as well. Thanks!

**Comment by davincho** [comment]
OK no worries. I am going to update my pull request this evening.

**Comment by kentcdodds** [comment]
Awesome! Thank you so much for working on this :+1: This is definitely needed!

**Comment by davincho** [comment]
@kentcdodds anything else you would like to include in this PR?

**Comment by kentcdodds** [comment]
Nope! It's perfect! Thanks!

**Pull request merged**